gapis/resolve/dependencygraph2/graph_visualization/graph_visualization.go

```diff
@@ -110,14 +123,15 @@ func GetGraphVisualizationFromCapture(ctx context.Context, p *path.Capture, form
 		return []byte{}, err
 	}
 
-	currentGraph, err := createGraphFromDependencyGraph(dependencyGraph)
-	currentGraph.removeNodesWithZeroDegree()
-	output := ""
+	currentGraph, err := createGraphFromDependencyGraph(ctx, dependencyGraph)
```

Comment:
It would be more canonical go with:

```go
	currentGraph, err := createGraphFromDependencyGraph(ctx, dependencyGraph)
	if err != nil {
		return []byte{}, err
        }
	currentGraph.joinNodesByFrame()
	currentGraph.joinNodesWithZeroDegree()
	output := []byte{}
	if format == service.GraphFormat_PBTXT {
		output = currentGraph.getGraphInPbtxtFormat()
	} else if format == service.GraphFormat_DOT {
		output = currentGraph.getGraphInDotFormat()
	}

	return output, nil
```

Reply:
You are right, I added the error checking.